### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by zhenyu-daniel** [pxpxkao/FinCausal-2020#1]
Hi Pei Wei,

I am doing NLP research now. I found you model is amazing. I run the model with the original full training dataset with the same hyper parameter on the paper. But the performance is way below the numbers claimed on the paper. Could you kindly suggest what am I missing? 

Sincerely,
Zhenyu

**Comment by soummyaah** [comment]
Hi @pxpxkao I am facing the same issue. I used the same hyper params mentioned in the paper with the original training dataset and am getting far lower values than expected. Could you help guide me please?

@zhenyu-daniel Did you resolve this issue?

Thanks

**Comment by pxpxkao** [comment]
Hello @soummyaah

Here's the message I replied to Zhenyu a few weeks ago, however, I didn't hear back from him, so I'm not sure if he solved the problem.

> Dear Zhenyu,
>
> Thank you for reading our work.
> Could you give me some more details about your training and testing procedures and the scores of the results? First, what testing dataset are you using to test the final results? Because predictions could be unpredictable based on different datasets. Second, did you modify any codes or parameters in the GitHub repo? If so, you could try to change it back and test it first. Third, the hardware you are using may be another reason for the changing results. Lastly, what are the results of your evaluation? How low is it? Is it far away from the results of the baseline BERT model that we said in the paper?
> To reproduce our results, you simply need to change the NUM_EPOCHS of “run.sh” from 1 to 4 and add or delete some flags, like --add_pos, then run it with the original full dataset. You should get a similar result when testing with the same blind test dataset. However, you can always use the model that we provided to evaluate your work. Thank you and hope you find the reasons soon!

If you're evaluating with another test set, you should probably analyze the dataset first to see if it has similar patterns with FinCausal-2020 since we didn't consider any transfer learning methods in this model.

Thank you.

**Comment by soummyaah** [comment]
Hi @pxpxkao,

I ran the model on the [FinCausal2020](https://github.com/yseop/YseopLab/tree/develop/FNP_2020_FinCausal/data) as well as the FinCausal2021 dataset which follows similar patterns as the 2020 dataset. From the 2020 dataset, I use the practice dataset containing 1109 examples for training data and the trial dataset with ground-truth containing 641 examples as the test dataset. I do not run the eval step. I understand that since we are training on lesser amount of data, there is an expected drop in the evaluation metrics. However, the drop is quite significant as listed below. I do not make any changes to the repository or the code.

The steps I followed were the following:
```
python3 preprocess.py data/fnp2020-fincausal2-task2-practice.csv train
python3 preprocess.py data/fnp2020-fincausal-task2-trial.csv test
cat data/train.txt data/test.txt | cut -d " " -f 2 | grep -v "^$"| sort | uniq > data/pos_tags.txt

export MAX_LENGTH=350
export BERT_MODEL=bert-base-cased
export DATA_DIR=data
export OUTPUT_DIR=fincausal
export BATCH_SIZE=4
export NUM_EPOCHS=4                  # Changed to 4
export SAVE_STEPS=3000
export SEED=1

CUDA_VISIBLE_DEVICES=0 python3 run.py \
--data_dir $DATA_DIR \
--labels $DATA_DIR/labels.txt \
--model_name_or_path $BERT_MODEL \
--output_dir $OUTPUT_DIR \
--max_seq_length  $MAX_LENGTH \
--num_train_epochs $NUM_EPOCHS \
--per_gpu_train_batch_size $BATCH_SIZE \
--save_steps $SAVE_STEPS \
--seed $SEED \
--do_train \
--do_predict \
--add_pos \

```

The results I obtain are these:
Number of exact matches: 29 out of 641
precision = 0.03717948717948718
recall = 0.0452418096723869
f1 = 0.04081632653061224
exact match = 0.0452418096723869
loss = 12.649067184071482


I also used the model provided on the repo, and used the steps provided in reproduce.sh, however I got the following results:
Number of exact matches: 3 out of 640
precision = 0.0015706806282722514
recall = 0.0046801872074883
f1 = 0.002352018816150529
exact match = 0.0046801872074883
loss = 467.83683720341435

I also get the following log messages:
```
Didn't find file eval_bio/added_tokens.json. We won't load it.
loading file eval_bio/vocab.txt
loading file None
loading file eval_bio/special_tokens_map.json
loading file eval_bio/tokenizer_config.json
loading weights file eval_bio/pytorch_model.bin
```

If I am missing any portion or step, please point me in the right direction.
